extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/3D.pulldown/Section Box Navigator.pushbutton/sectionbox_actions.py

```diff
@@ -0,0 +1,158 @@
+from pyrevit import revit, script, forms
+from pyrevit.framework import List, Math
+from pyrevit.compat import get_elementid_value_func
+from pyrevit import DB, UI
+
+get_elementid_value = get_elementid_value_func()
+
+logger = script.get_logger()
+
+
+def toggle(doc, datafilename):
+    """Toggle section box."""
+    current_view = doc.ActiveView
+    if not isinstance(current_view, DB.View3D):
+        return
+    current_view_id_value = get_elementid_value(current_view.Id)
+    sectionbox_active_state = current_view.IsSectionBoxActive
+
+    try:
+        view_boxes = script.load_data(datafilename)
+    except Exception:
+        view_boxes = {}
+
+    with revit.Transaction("Toggle SectionBox"):
+        if sectionbox_active_state:
+            try:
+                sectionbox = current_view.GetSectionBox()
+                if sectionbox:
+                    view_boxes[current_view_id_value] = revit.serialize(sectionbox)
+                    script.store_data(datafilename, view_boxes)
+                current_view.IsSectionBoxActive = False
+            except Exception as ex:
+                logger.error("Error saving section box: {}".format(ex))
+        else:
+            try:
+                if current_view_id_value in view_boxes:
+                    bbox_data = view_boxes[current_view_id_value]
+                    restored_bbox = revit.deserialize(bbox_data)
+                    current_view.SetSectionBox(restored_bbox)
+            except Exception as ex:
+                logger.error(
+                    "No saved section box found or failed to load: {}".format(ex)
+                )
+
+
+def hide(doc):
+    """Hide or Unhide section box."""
+    current_view = doc.ActiveView
+    if not isinstance(current_view, DB.View3D):
+        return
+    with revit.Transaction("Toggle SB visbility"):
+        current_view.EnableRevealHiddenMode()
+        view_elements = (
+            DB.FilteredElementCollector(doc, current_view.Id)
+            .OfCategory(DB.BuiltInCategory.OST_SectionBox)
+            .ToElements()
+        )
+        for sec_box in [
+            x for x in view_elements if x.CanBeHidden(current_view)
+        ]:
+            if sec_box.IsHidden(current_view):
+                current_view.UnhideElements(List[DB.ElementId]([sec_box.Id]))
+            else:
+                current_view.HideElements(List[DB.ElementId]([sec_box.Id]))
```

Comment:
```suggestion
            ids = List[DB.ElementId]([sec_box.Id])
            if sec_box.IsHidden(current_view):
                current_view.UnhideElements(ids)
            else:
                current_view.HideElements(ids)
```

But I wonder if it is better to collect all the ids to hide/unhide in two lists and then use `UnhideElements` and `HideElements` once. Can we do a simple benchmark here?